### PR TITLE
[proposal-draft] scaled-back MDX integration guide

### PR DESCRIPTION
**Pull request by sarah11918**
- New or updated content

Not to actually be merged! For preview purposes only!

This is a WIP, new proposal to replace the MDX integrations guide, once most of the content is moved to Docs.

This retains the Integration Guide structure, and strips down the "Usage" section to links to our Markdown/MDX content. It therefore focuses on Installation and Configuration.

I think there's still some duplication/wording to hash out. But initial concerns are: is this "enough" to be a usable README to stand on its own, and point to docs for most things usage/layout etc.?

Eggregious use of `/integrations/` to host this not-intended-where-it-is preview:  https://deploy-preview-2060--astro-docs-2.netlify.app/en/integrations/mdx/

### CODE REVIEW

File: src/pages/en/integrations/mdx.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    // Applied to .mdx files only
+    rehypePlugins: [rehypeMinify],
+  })],
+}
+```
+
+### extendPlugins
+
+When adding your own plugins, [GitHub-flavored Markdown](https://github.com/remarkjs/remark-gfm) and [Smartypants](https://github.com/silvenon/remark-smartypants) are removed. You can preserve these defaults with the `extendDefaultPlugins` flag.
+
```
Comment (sarah11918) [inline src/pages/en/integrations/mdx.md:118]:
```suggestion

**When adding your own plugins, these two default plugins are removed.** You can preserve these defaults with the [`markdown.extendDefaultPlugins` flag](#extendplugins) or with `extendPlugins` in your `mdx` integration config.

You can browse [awesome-remark](https://github.com/remarkjs/awesome-remark) and [awesome-rehype](https://github.com/rehypejs/awesome-rehype) for popular plugins. See each plugin's own README for specific installation instructions.

#### Inherited Plugins from `markdown` config

By default, Astro's MDX integration inherits all [remark](https://github.com/withastro/astro/tree/main/packages/integrations/mdx/#remarkplugins) and [rehype](https://github.com/withastro/astro/tree/main/packages/integrations/mdx/#rehypeplugins) plugins from [the `markdown` option in your Astro config](/en/guides/markdown-content/#markdown-plugins).

Any additional plugins you apply in your MDX config will be applied *after* your configured Markdown plugins, and will apply only to `.mdx` files.

This example applies [`remark-toc`](https://github.com/remarkjs/remark-toc) to Markdown *and* MDX, and [`rehype-minify`](https://github.com/rehypejs/rehype-minify) to MDX only:

```js title="astro.config.mjs"
import { defineConfig } from 'astro/config';
import remarkToc from 'remark-toc';
import rehypeMinify from 'rehype-minify';
export default {
  markdown: {
    // Applied to .md and .mdx files
    remarkPlugins: [remarkToc],
  },
  integrations: [mdx({
    // Applied to .mdx files only
    rehypePlugins: [rehypeMinify],
  })],
}
```

### extendPlugins

When adding your own plugins, [GitHub-flavored Markdown](https://github.com/remarkjs/remark-gfm) and [Smartypants](https://github.com/silvenon/remark-smartypants) are removed. You can preserve these defaults in both Markdown and MDX files with the `markdown.extendDefaultPlugins` flag.

To control extending plugins only in MDX, you can configure `.extendPlugins` in your `mdx` integration directly.

```

File: src/pages/en/integrations/mdx.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    // Applied to .md and .mdx files
+    remarkPlugins: [remarkToc],
+  },
+  integrations: [mdx({
+    // Applied to .mdx files only
+    rehypePlugins: [rehypeMinify],
+  })],
+}
+
+**Type:** `'markdown' | 'astroDefaults' | false`
```
Comment (sarah11918) [inline src/pages/en/integrations/mdx.md:114]:
```suggestion
### extendPlugins
When adding your own plugins, [GitHub-flavored Markdown](https://github.com/remarkjs/remark-gfm) and [Smartypants](https://github.com/silvenon/remark-smartypants) are removed. You can preserve these defaults in Markdown and MDX files with the `markdown.extendDefaultPlugins` flag.

To control extending plugins only in MDX, you can configure `.extendPlugins` in your `mdx` integration directly.

**Type:** `'markdown' | 'astroDefaults' | false`
```

File: src/pages/en/integrations/mdx.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    rehypePlugins: [rehypeMinify],
+  })],
+}
+```
+
+### extendPlugins
+
+When adding your own plugins, [GitHub-flavored Markdown](https://github.com/remarkjs/remark-gfm) and [Smartypants](https://github.com/silvenon/remark-smartypants) are removed. You can preserve these defaults in Markdown and MDX files with the `markdown.extendDefaultPlugins` flag.
+
+To control extending plugins only in MDX, you can configure `.extendPlugins` in your `mdx` integration directly.
```
Comment (sarah11918) [inline src/pages/en/integrations/mdx.md:119]:
```suggestion
To control extending plugins only in MDX, you can configure `.extendPlugins` in your `mdx` integration directly.

```

**Review by delucis — COMMENTED** [review]
This is great @sarah11918! Several really clear improvements and seeing this further encourages me that the restructuring in #2054 is on the right track.

Just a couple of notes:

- ➕ I might add a minimal mention (maybe in “Usage”?) about MDX supporting similar features to Markdown (layouts, frontmatter support by default — which isn’t the case in vanilla MDX).

- 🖌️ We can improve the structure of the “Configuration” section. The subheads currently are: “Markdown Plugins,” “extendPlugins,” and “recmaPlugins.” The latter two are actual configuration options, but the first isn’t which feels a little off to me.

    We might want a clearer distinction between a reference-y “Configuration Options,” which lists each of the options you can pass to `mdx({ ... })` (and should include _all_ those options, so also `rehypePlugins` and `remarkPlugins` if only briefly) and separate the more explanatory material somehow.

File: src/pages/en/integrations/mdx.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+"files.associations": {
+    "*.mdx": "markdown"
+}
+```
+
+## Usage
+
+Visit the [MDX docs](https://mdxjs.com/docs/what-is-mdx/) to learn about using MDX as a content authoring solution.
+
+In Astro, you can [add MDX pages to your project](/en/guides/markdown-content/#markdown-and-mdx-pages) by adding `.mdx` files within your `src/pages/` directory. You can also [import `.mdx` files](https://docs.astro.build/en/guides/markdown-content/#importing-markdown) into `.astro` files.
```
Comment (sarah11918) [inline src/pages/en/integrations/mdx.md:74]:
```suggestion
With the Astro MDX integration, you can [add MDX pages to your project](/en/guides/markdown-content/#markdown-and-mdx-pages) by adding `.mdx` files within your `src/pages/` directory. You can also [import `.mdx` files](https://docs.astro.build/en/guides/markdown-content/#importing-markdown) into `.astro` files. 

Astro's MDX integration adds extra features to standard MDX, allowing you to use most of Astro's built-in Markdown features like a frontmatter `layout` property, a setting for draft pages, as well as default support for frontmatter variables.
```

File: src/pages/en/integrations/mdx.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+```json title=".vscode/settings.json"
+"files.associations": {
+    "*.mdx": "markdown"
+}
+```
+
+## Usage
+
+Visit the [MDX docs](https://mdxjs.com/docs/what-is-mdx/) to learn about using MDX as a content authoring solution.
```
Comment (sarah11918) [inline src/pages/en/integrations/mdx.md:72]:
```suggestion
```

File: src/pages/en/integrations/mdx.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+import { defineConfig } from 'astro/config';
+import mdx from '@astrojs/mdx';
+
+export default defineConfig({
+  // ...
+  integrations: [mdx()],
+});
+```
+
+Finally, restart the dev server.
```
Comment (delucis) [inline src/pages/en/integrations/mdx.md:58]: I guess we can remove this now that this happens for you?

```suggestion
```

File: src/pages/en/integrations/mdx.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+import remarkToc from 'remark-toc';
+
+export default {
+  integrations: [mdx({
+    remarkPlugins: [remarkToc],
+  })],
+}
+```
+
+### rehypePlugins
```
Comment (delucis) [inline src/pages/en/integrations/mdx.md:101]:
```suggestion
### `rehypePlugins`
```

File: src/pages/en/integrations/mdx.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+export default {
+  integrations: [mdx({
+    rehypePlugins: [rehypeMinifyHtml],
+  })],
+}
+```
+
+
+### extendPlugins
```
Comment (delucis) [inline src/pages/en/integrations/mdx.md:121]:
```suggestion
### `extendPlugins`
```

File: src/pages/en/integrations/mdx.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+```js
+// astro.config.mjs
+import rehypeMinifyHtml from 'rehype-minify';
+
+export default {
+  integrations: [mdx({
+    rehypePlugins: [rehypeMinifyHtml],
+  })],
+}
+```
```
Comment (delucis) [inline src/pages/en/integrations/mdx.md:118]: We should change this! Didn’t notice it was also here when I made https://github.com/withastro/docs/pull/2025, but this does not work. We can probably copy over the changes I made there.

File: src/pages/en/integrations/mdx.md [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+
+
+### extendPlugins
+
+**Type:** `'markdown' | 'astroDefaults' | false`
+
+**Default:** `'markdown'`
```
Comment (delucis) [inline src/pages/en/integrations/mdx.md:125]: Gut instinct is that we probably want a short generic description of `extendPlugins` before diving into the specific options. We don’t show type/default for the other options and I think the subheadings make this fairly obvious so maybe we can skip those?

```suggestion
You can customize how MDX files inherit your project’s existing Markdown configuration using the `extendPlugins` option.
```

**Review by delucis — COMMENTED** [review]
Looking good @sarah11918! Made some suggestions, but I’d say after this, this is probably ready for a PR to the README 🙌

File: src/pages/en/integrations/mdx.md [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+
+export default {
+  integrations: [mdx({
+    rehypePlugins: [rehypeMinifyHtml],
+  })],
+}
+```
+
+
+### `extendPlugins`
```
Comment (sarah11918) [inline src/pages/en/integrations/mdx.md:120]:
```suggestion
### remarkRehype
Options for `remarkRehype`. See [markdown.remarkRehype](https://docs.astro.build/en/reference/configuration-reference/#markdownremarkrehype).

When `extendPlugins` is set to `'markdown'` (default), this config is extended with `remarkRehype` from Astro's default markdown config.

### `extendPlugins`
```
This is being PR'd as a new config option. Adding it here as a placeholder so I include it in my edit.

File: src/pages/en/integrations/mdx.md [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+```js
+// astro.config.mjs
+import rehypeMinifyHtml from 'rehype-minify';
+
+export default {
+  integrations: [mdx({
+    rehypePlugins: [rehypeMinifyHtml],
+  })],
+}
+```
```
Comment (sarah11918) [inline src/pages/en/integrations/mdx.md:117]:
```suggestion
This example applies the [`rehype-accessible-emojis`](https://www.npmjs.com/package/rehype-accessible-emojis) plugin to `.mdx` files. To customize plugin inheritance from your Markdown config or Astro's defaults, [see the `extendPlugins` option](#extendplugins).

```js
// astro.config.mjs
import rehypeAccessibleEmojis from 'rehype-accessible-emojis';

export default {
  integrations: [mdx({
    rehypePlugins: [rehypeAccessibleEmojis],
  })],
}
```
```

File: src/pages/en/integrations/mdx.md [hunk 13]
```diff
@@ -0,0 +1,218 @@
...
+```js
+// astro.config.mjs
+import rehypeAccessibleEmojis from 'rehype-accessible-emojis';
+
+export default {
+  integrations: [mdx({
+    rehypePlugins: [rehypeAccessibleEmojis],
+  })],
+}
+
```
Comment (sarah11918) [inline src/pages/en/integrations/mdx.md:117]:
``````suggestion
```

``````

Comment (sarah11918) [reply src/pages/en/integrations/mdx.md:117]: missing end code block syntax

File: src/pages/en/integrations/mdx.md [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+};
+```
+This inherits the configuration of [`markdown.remarkRehype`](https://docs.astro.build/en/reference/configuration-reference/#markdownremarkrehype). This behavior can be changed by configuring `extendPlugins`.
+
+### `extendPlugins`
+
+You can customize how MDX files inherit your project’s existing Markdown configuration using the `extendPlugins` option.
+
+
+
```
Comment (sarah11918) [inline src/pages/en/integrations/mdx.md:144]:
```suggestion
```

File: src/pages/en/integrations/mdx.md [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+    "*.mdx": "markdown"
+}
+```
+
+## Usage
+
+
+With the Astro MDX integration, you can [add MDX pages to your project](/en/guides/markdown-content/#markdown-and-mdx-pages) by adding `.mdx` files within your `src/pages/` directory. You can also [import `.mdx` files](https://docs.astro.build/en/guides/markdown-content/#importing-markdown) into `.astro` files. 
+
+Astro's MDX integration adds extra features to standard MDX, including support for Markdown-style frontmatter. This allows you to use most of Astro's built-in Markdown features like a frontmatter `layout` property and a setting for draft pages.
```
Comment (sarah11918) [inline src/pages/en/integrations/mdx.md:74]:
```suggestion
Astro's MDX integration adds extra features to standard MDX, including Markdown-style frontmatter. This allows you to use most of Astro's built-in Markdown features like a [special frontmatter `layout` property](https://docs.astro.build/en/guides/markdown-content/#frontmatter-layout) and a [property for marking a page as a draft](https://docs.astro.build/en/guides/markdown-content/#draft-pages).
```

File: src/pages/en/integrations/mdx.md [hunk 16]
```diff
@@ -0,0 +1,217 @@
...
+## Usage
+
+
+With the Astro MDX integration, you can [add MDX pages to your project](/en/guides/markdown-content/#markdown-and-mdx-pages) by adding `.mdx` files within your `src/pages/` directory. You can also [import `.mdx` files](https://docs.astro.build/en/guides/markdown-content/#importing-markdown) into `.astro` files. 
+
+Astro's MDX integration adds extra features to standard MDX, including Markdown-style frontmatter. This allows you to use most of Astro's built-in Markdown features like a [special frontmatter `layout` property](https://docs.astro.build/en/guides/markdown-content/#frontmatter-layout) and a [property for marking a page as a draft](https://docs.astro.build/en/guides/markdown-content/#draft-pages).
+
+See how MDX works in Astro with examples in our [MD/MDX guide](https://docs.astro.build/en/guides/markdown-content/).
+
+Visit the [MDX docs](https://mdxjs.com/docs/what-is-mdx/) to learn about using MDX as a content authoring solution.
```
Comment (sarah11918) [inline src/pages/en/integrations/mdx.md:78]:
```suggestion
Visit the [MDX docs](https://mdxjs.com/docs/what-is-mdx/) to learn about using standard MDX features.
```

File: src/pages/en/integrations/mdx.md [hunk 17]
```diff
@@ -0,0 +1,217 @@
...
+
+export default {
+  integrations: [mdx({
+    rehypePlugins: [rehypeAccessibleEmojis],
+  })],
+}
+```
+
+
+### remarkRehype
```
Comment (sarah11918) [inline src/pages/en/integrations/mdx.md:122]:
```suggestion
### `remarkRehype`
```

**Comment by delucis** [comment]
We did it! 👋 🥲

**Pull request closed without merging**